.github/workflows/test.yml

```diff
@@ -23,4 +23,4 @@ jobs:
         go-version: "1.22"
         cache: true
     - name: Test
-      run: make GO_TAGS="nodocker" test
+      run: CGO_LDFLAGS="-ld_classic $CGO_LDFLAGS" make GO_TAGS="nodocker" test
```

Comment:
Why is this change needed?

Reply:
@tpaschalis, without `CGO_LDFLAGS="-ld_classic` build fails on mac.

Reply:
Can you describe what this does and share a link to some docs where we can read more about this? I'm not familiar with this option.

Reply:
Good question, @thampiotr 👍 

When this PR was created, the build on macOS returned linker errors. This flag tells CI to use a classic linker, which is supposed to be more compatible. It solved a problem back then. 

Now I see that Apple is deprecating `-ld_classic` https://developer.apple.com/documentation/xcode-release-notes/xcode-16-release-notes#Deprecations-in-Xcode-16-Beta. 

I need to check if this is still required for our tests.

Some additional information about `ld_classic` can be found here:

- https://forums.developer.apple.com/forums/thread/715385
- https://www.unix.com/man-page/osx/1/ld_classic

Reply:
JFYI: I tried to remove it, but the build failed.